File: engines/php/src/CleaningEstimate.php
<?php

declare(strict_types=1);

namespace WindowCleaning;

/**
 * Detailed cost breakdown for a commercial window cleaning job.
 * All monetary values are in CAD.
 */
final class CleaningEstimate
{
    public function __construct(
        public readonly float $labourCost,
        public readonly float $materialsCost,
        public readonly float $subtotal,
        public readonly float $marginAmount,
        public readonly float $finalPrice,
        public readonly float $perWindowCost,
    ) {}

    public function toArray(): array
    {
        return [
            'labour_cost' => $this->labourCost,
            'materials_cost' => $this->materialsCost,
            'subtotal' => $this->subtotal,
            'margin_amount' => $this->marginAmount,
            'final_price' => $this->finalPrice,
            'per_window_cost' => $this->perWindowCost,
        ];
    }
}

File: engines/java/src/main/java/io/github/davecookvectorlabs/windowcleaning/CleaningEstimate.java
package io.github.davecookvectorlabs.windowcleaning;

/**
 * Detailed cost breakdown for a commercial window cleaning job.
 * All monetary values are in CAD.
 */
public class CleaningEstimate {
    private final double labourCost;
    private final double materialsCost;
    private final double subtotal;
    private final double marginAmount;
    private final double finalPrice;
    private final double perWindowCost;

    public CleaningEstimate(double labourCost, double materialsCost, double subtotal,
                            double marginAmount, double finalPrice, double perWindowCost) {
        this.labourCost = labourCost;
        this.materialsCost = materialsCost;
        this.subtotal = subtotal;
        this.marginAmount = marginAmount;
        this.finalPrice = finalPrice;
        this.perWindowCost = perWindowCost;
    }

    public double getLabourCost() { return labourCost; }
    public double getMaterialsCost() { return materialsCost; }
    public double getSubtotal() { return subtotal; }
    public double getMarginAmount() { return marginAmount; }
    public double getFinalPrice() { return finalPrice; }
    public double getPerWindowCost() { return perWindowCost; }

    @Override
    public String toString() {
        return String.format(
            "CleaningEstimate{labour=%.2f, materials=%.2f, subtotal=%.2f, margin=%.2f, total=%.2f, perWindow=%.2f}",
            labourCost, materialsCost, subtotal, marginAmount, finalPrice, perWindowCost
        );
    }
}
